src/emsarray/operations/cache.py

```diff
@@ -0,0 +1,72 @@
+"""
+Operations for making a triangular mesh out of the polygons of a dataset.
+"""
+from __future__ import annotations
+
+import hashlib
+import inspect
+import pickle
+from typing import Any
+
+import xarray
+
+
+def hash_attributes(hash: hashlib._Hash, attribute_dict: xarray.Variable) -> None:
```

Comment:
The pickled representation of an `xarray.Variable` includes information we don't need, such as the filename of the opened dataset included in the `DataArray.encoding` attribute, and also the variable data. This method should take a plain `dict` of attributes such as `DataArray.attrs` not an `xarray.Variable`. 

```python
def hash_attributes(hash: "hashlib._Hash", attributes: dict) -> None:
    ...
```

```python
def hash_geometry(self, hash):
    for geometry_name in self.get_all_geometry_names():
        data_array = self.dataset[geometry_name]
        # ... hashing of the name, dtype, data, etc
        hash_attributes(data_array.attrs)
```

Additionally this allows Convention subclasses to hash select global attributes, such as:

```python
class CFGrid1D(Convention):
    def hash_geometry(self, hash: "hashlib._Hash") -> None:
        super().hash_geometry(hash)

        global_geometry_attributes = ['Conventions']
        hash_attributes(hash, {name: self.dataset.attrs[name] for name in global_geometry_attributes})
```

Reply:
Updated so were only serialising the data_array.attrs as suggested.